src/onboarding2/Create/Create.js

```diff
@@ -54,73 +70,135 @@ function useTemplateState() {
     } = loadTemplateState()
 
     if (templateId) {
+      updateTemplateScreen(templateId, templateScreenIndex)
       setTemplateData(templateData)
-      setTemplateId(templateId)
-      setTemplateScreenIndex(templateScreenIndex)
     }
-  }, [])
+  }, [updateTemplateScreen])
 
-  const [template, screens] = useMemo(() => {
-    const template = templates.find(template => template.id === templateId)
-    return [
-      template,
-      (template && template.template && template.template.screens) || [],
-    ]
-  }, [templateId])
+  // Save the template state
+  useEffect(() => {
+    if (template && template.id) {
+      saveTemplateState({
+        templateData,
+        templateId: template.id,
+        templateScreenIndex,
+      })
+    }
+  }, [templateData, template, templateScreenIndex])
 
-  const updateScreenIndex = useCallback(
+  const relativeScreen = useCallback(
     diff => {
       const updatedScreenIndex = templateScreenIndex + diff
 
       // Back to the templates selection
-      if (updatedScreenIndex < 0) {
+      if (updatedScreenIndex < 0 || !template) {
         setTemplateData({})
-        setTemplateId(null)
-        setTemplateScreenIndex(-1)
+        updateTemplateScreen(null, -1)
         return
       }
 
-      setTemplateScreenIndex(screenIndex =>
-        Math.min(screens.length - 1, updatedScreenIndex)
+      updateTemplateScreen(
+        template.id,
+        Math.min(template.screens.length, updatedScreenIndex)
```

Comment:
We should leave a comment here mentioning that `templates.screen.length` is acceptable because it's the "deployment" screen.

Or if we move the state up, we may be able to just use the normal length of screens and invoke a callback when we hit the deployment phase